modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RankEvalTestHelper.java

```diff
@@ -42,4 +49,20 @@ public static XContentParser roundtrip(ToXContentToBytes testItem) throws IOExce
         XContentParser itemParser = XContentHelper.createParser(shuffled.bytes());
         return itemParser;
     }
+
+    public static void testHashCodeAndEquals(Object testItem, Object mutation, Object secondCopy) {
```

Comment:
Yes, good idea. This will be a bit of an effort though, e.g. using this in AbstractQueryTestCase currently is not possible because we don't have the mutation methods for the queries there yet. Also, after reading https://www.petrikainulainen.net/programming/unit-testing/3-reasons-why-we-should-not-use-inheritance-in-our-tests recently, I started to change my mind a bit about introducing too much inheritance hierarchy, dependecies etc... to avoid any kind of code duplication. Maybe in this case makes sense to pull this up into a separate helper, that would be another issue though.